api/src/Validator/ContentNode/AssertAttachedToRootValidator.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Validator\ContentNode;
+
+use App\Entity\ContentNode;
+use App\Entity\ContentNode\DefaultLayout;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+class AssertAttachedToRootValidator extends ConstraintValidator {
+    public function validate($value, Constraint $constraint): void {
+        if (!$constraint instanceof AssertAttachedToRoot) {
+            throw new UnexpectedTypeException($constraint, AssertAttachedToRoot::class);
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        $node = $this->context->getObject();
+        if (!$node instanceof ContentNode) {
+            throw new UnexpectedValueException($node, ContentNode::class);
+        }
+
+        if (!$node instanceof DefaultLayout) {
+            return;
+        }
```

Comment:
i would prefer that this validation is only attached to the DefaultLayout, and this Validator throws if it is attached to another Class.

Reply:
I'm not sure if this is the favorable path, as it requires https://github.com/ecamp/ecamp3/pull/4071#discussion_r1412830114

Which way should I go?